scm-ui/ui-webapp/src/repos/branches/containers/DeleteBranch.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020-present Cloudogu GmbH and Contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+import React, { FC, useState } from "react";
+import { useHistory } from "react-router-dom";
+import { useTranslation } from "react-i18next";
+import { Branch, Link, Repository } from "@scm-manager/ui-types";
+import { apiClient, ConfirmAlert, DeleteButton, ErrorNotification, Level } from "@scm-manager/ui-components";
+
+type Props = {
+  repository: Repository;
+  branch: Branch;
+};
+
+const DeleteBranch: FC<Props> = ({ repository, branch }: Props) => {
+  const [showConfirmAlert, setShowConfirmAlert] = useState(false);
+  const [error, setError] = useState<Error | undefined>();
+  const [t] = useTranslation("repos");
+  const history = useHistory();
+
+  const deleteBranch = () => {
+    apiClient
+      .delete((branch._links.delete as Link).href)
+      .then(() => history.push(`/repo/${repository.namespace}/${repository.name}/branches/`))
+      .catch(setError);
+  };
+
+  if (!branch._links.delete) {
+    return null;
+  }
+
+  const confirmAlert = (
```

Comment:
Shouldnt the check for "showConfirmAlert" be here so the ConfirmAlert instance doesnt get created if the flag is false ?

Reply:
Had it first like you suggests but i wondered if this really make a difference. I will change it nonetheless :+1: